pkg/asset/cluster/tfvars.go

```diff
@@ -672,6 +675,46 @@ func (t *TerraformVariables) Generate(parents asset.Parents) error {
 			Filename: TfPlatformVarsFileName,
 			Data:     data,
 		})
+	case powervs.Name:
+		masters, err := mastersAsset.Machines()
+		if err != nil {
+			return err
+		}
+
+		// Get CISInstanceCRN from InstallConfig metadata
+		crn, err := installConfig.PowerVS.CISInstanceCRN(ctx)
+		if err != nil {
+			return err
+		}
+
+		masterConfigs := make([]*powervsprovider.PowerVSMachineProviderConfig, len(masters))
+		for i, m := range masters {
+			masterConfigs[i] = m.Spec.ProviderSpec.Value.Object.(*powervsprovider.PowerVSMachineProviderConfig)
+		}
+
+		data, err = powervstfvars.TFVars(
+			powervstfvars.TFVarsSources{
+				MasterConfigs:        masterConfigs,
+				Region:               installConfig.Config.Platform.PowerVS.Region,
+				Zone:                 installConfig.Config.Platform.PowerVS.Zone,
+				APIKey:               os.Getenv("IC_API_KEY"),
```

Comment:
You shouldn't set an environment variable to retrieve that value later in the code... This API key is in your session struct. The session struct should be stored in the install config metadata and then you can use that to retrieve the value here.